packages/project-access/src/project/cap.ts

```diff
@@ -184,13 +185,36 @@ export async function getCapEnvironment(capProjectPath: string): Promise<CdsEnvi
 }
 
 /**
- * Load CAP CDS module for a project based on its root.
+ * Load CAP CDS module. First attempt loads @sap/cds for a project based on its root.
+ * Second attempt loads @sap/cds from global installed @sap/cds-dk.
+ * Throws error if module could not be loaded.
  *
  * @param capProjectPath - project root of a CAP project
  * @returns - CAP CDS module for a CAP project
  */
 async function loadCdsModuleFromProject(capProjectPath: string): Promise<CdsFacade> {
-    const module = await loadModuleFromProject<CdsFacade | { default: CdsFacade }>(capProjectPath, '@sap/cds');
+    let module: CdsFacade | { default: CdsFacade } | undefined;
+    let loadProjectError;
+    let loadError;
+    try {
+        // First approach, load @sap/cds from project
+        module = await loadModuleFromProject<CdsFacade | { default: CdsFacade }>(capProjectPath, '@sap/cds');
```

Comment:
```suggestion
       const versions = await getCdsVersionInfo(capProjectPath);
        if (!versions.home) {
        const CdsPathCache = versions.home;
        module = await loadModuleFromProject<CdsFacade | { default: CdsFacade }>(CdsPathCache, '@sap/cds');
```
Can't we rely `cds --version` to resolve the cds location (assuming home is returned in older versions of the cli ??)  and use that path for `loadModuleFromProject`

Reply:
We can rely on `cds --version`, but it is rather slow to execute the command each time we access the cds module. For instance, we use `getCapProjectType()` which calls `getCapCustomPaths()` -> `getCapEnvironment()` -> `loadCdsModuleFromProject()` many times when searching all projects in workspace, we use `getCapModelAndServices()` and `readCapServiceMetadataEdmx()` in design time after almost every step, so calling a terminal command each time before would have a very negative impact on performance. TI thought about creating a cache `Map<string, string>` where we store project root -> path to cds, but we also need to handle following scenario:
1. `@sap/cds` is not installed locally in the project
2. our tools access it and cache the path to global installation
3. `@sap/cds` is installed in the project
Not sure how we can detect this and invalidate the cache. 

This is why I went for the solution on this PR:
1. Try to load local `@sap/cds` from project
2. If it exists, it will be in `require` cache and loading is fast
3. If it does not exist, try to load global installed `@sap/cds-dk` (path is also cached)

Let me know if you find a better solution.

Reply:
Sounds good. Thanks